### PR TITLE
[docs] Add <Tabs/> component plugin

### LINKED ISSUE #10189: [docs] Differentiate new and legacy notification methods in unversioned pages
The documentation is not consistent on notification setup and uses an (I assume) old command "createChannelAndroidAsync".

The old / not updated part of the documentation is here, see the last part of the code:
https://docs.expo.io/push-notifications/push-notifications-setup/

This is the old code, bold the old stuff:
```
  if (Platform.OS === 'android') {
    Notifications.**createChannelAndroidAsync**('default', {
      name: 'default',
      sound: true,
      **priority: 'max',**
      vibrate: [0, 250, 250, 250],
    });
  }
```
It should be this, I think according to other parts of the docs and what I have found works:
```
  if (Platform.OS === 'android') {
    Notifications.**setNotificationChannelAsync**('default', {
      name: 'default',
      **importance: Notifications.AndroidImportance.MAX,**
      vibrationPattern: [0, 250, 250, 250],
      lightColor: '#FF231F7C',
    });

```

### PR DESCRIPTION
**Pull request by barthap**
# Why

Resolves #10189 and it's direct answer to https://github.com/expo/expo/issues/10189#issuecomment-692996355
I haven't found support for Tabs in current Expo docs.

# How

Added `<Tabs/>` component to docs plugins. Updated notifications guide to use the component. Demo GIF's below.

Usage:
```js

import { Tab, Tabs } from '~/components/plugins/Tabs';

// ...

<Tabs>
  <Tab label="First tab">First tab content</Tab>
  <Tab label="Second tab">Lorem ipsum</Tab>
</Tabs>

// or another way
<Tabs tabs={['First tab', 'Second tab']}>
  <Tab> First tab content</Tab>
  <Tab> Second tab content</Tab>
</Tabs>
```

The tabs accept any markup inside, however it's not recommended to put headings inside, because they won't work properly with #10063, unless those headings are explicitly hidden from the sidebar.

# Test Plan

Renders correctly in `yarn dev`. On mobile, when there are more tabs, they are split into rows.

---

tabs1

Demo below has random text copied, please ignore this 😅 

tabs2

### CODE REVIEW

File: docs/components/plugins/Tabs.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+export const Tab = ({ children }) => children;
+
+/**
+ * @example
+ * <Tabs>
+ *   <Tab label="Tab1">Tab 1 content...</Tab>
+ *   <Tab label="Tab2">Tab 2 content...</Tab>
+ * </Tabs>
+ */
+
```
Comment (barthap) [inline docs/components/plugins/Tabs.js:64]:
```suggestion
```

File: docs/components/plugins/Tabs.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        {children.map(child => {
+          const { label } = child.props;
+          return (
+            <TabButton activeTab={activeTab} key={label} label={label} onClick={setActiveTab} />
+          );
+        })}
+      </ol>
+      <div className={STYLES_TAB_CONTENT}>
+        {children.map(child => {
+          if (child.props.label !== activeTab) return;
```
Comment (barthap) [inline docs/components/plugins/Tabs.js:80]:
```suggestion
          if (child.props.label !== activeTab) {
            return;
          }
```

**Comment by bbarthec** [comment]
Why not to use some 3rd party library? For example: https://github.com/reactjs/react-tabs?

**Comment by barthap** [comment]
I just wanted to keep it as simple and lightweight as possible. IMO most of these libraries look like an overkill. My implementation is only ~80 lines of code and IMO it's easier to use than the one you linked. But if you prefer to, I can just replace my implementation with that library.

**Review by bbarthec — APPROVED** [review]
I'm not a fun of this linked library as well. I just wanted to know whether you've checked what's already there is the wild world of internet 😉

**Comment by byCedric** [comment]
I think Evan already added [`@react/tabs`](https://github.com/expo/expo/blob/master/docs/package.json#L23) for the [authentication guides](https://docs.expo.io/guides/authentication/#coinbase).

Screenshot 2020-09-18 at 02 06 35

We might get away with just re-exporting these components somewhere from `~/components/plugins/tabs` 😄

**Comment by barthap** [comment]
Indeed, I didn't notice that, looks like AuthSession components might be made more universal:

https://github.com/expo/expo/blob/7db65dee00cc9fb0395fd8a766deb3cb8a20131c/docs/components/plugins/AuthSessionElements.js#L121-L166

The question is which style we want? I don't see auth tabs being affected by latest styling update, IMO they're looking cool as they are

**Comment by bbarthec** [comment]
@barthap, indeed, your approach seems easier to be used IMO. To keep things consistent you can now update this other component with the idea of yours and use the single component in all places 😉

**Comment by barthap** [comment]
To ensure I understand correctly, you mean:
- Rework tabs API to look my way
- Leave CSS styling as it was in [Authentication Guide](https://docs.expo.io/guides/authentication/#dropbox)

But shall we keep using `@reach/tabs` library or switch to my plain solution? 🤔

**Comment by bbarthec** [comment]
@barthap, if refactoring this API is not going to take long let's do this. Let's keep `@reach/tabs` 👍

**Comment by barthap** [comment]
Ok, I've updated it, now it:
- Has _Authentication Guide_ look
- Uses `@reach/tabs` library
- Works using both mine and little modified Evan's syntax

File: docs/pages/guides/authentication.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ title: Authentication
 import PlatformsSection from '~/components/plugins/PlatformsSection';
 import InstallSection from '~/components/plugins/InstallSection';
 import TableOfContentSection from '~/components/plugins/TableOfContentSection';
-import { SocialGrid, SocialGridItem, CreateAppButton, AuthMethodTabSwitcher, ImplicitTab, AuthMethodTab, AuthCodeTab } from '~/components/plugins/AuthSessionElements';
+import { SocialGrid, SocialGridItem, CreateAppButton, AuthMethodTabSwitcher, AuthMethodTab } from '~/components/plugins/AuthSessionElements';
```
Comment (bbarthec) [inline docs/pages/guides/authentication.md:8]:
```suggestion
import { SocialGrid, SocialGridItem, CreateAppButton } from '~/components/plugins/AuthSessionElements';
import { Tab, Tabs } from '~/components/plugins/Tabs';
```
Why not ot use these Tabs of your directly? 🤔

File: docs/components/plugins/AuthSessionElements.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import { Tab, TabList, TabPanel, TabPanels, Tabs } from '@reach/tabs';
 import * as React from 'react';
 import { css } from 'react-emotion';
-import * as Constants from '~/constants/theme';
+
+import { Tab, Tabs } from './Tabs';
```
Comment (bbarthec) [inline docs/components/plugins/AuthSessionElements.js:4]:
```suggestion
```

File: docs/components/plugins/AuthSessionElements.js [hunk 5]
```diff
@@ -102,64 +102,5 @@ export function SocialGridItem({ title, protocol = [], image, href }) {
...
-            </AuthMethodTabButton>
-          );
-        })}
-      </TabList>
-      <TabPanels style={{ paddingTop: 8 }}>{children}</TabPanels>
-    </Tabs>
-  );
-}
+export const AuthMethodTab = Tab;
+export const AuthMethodTabSwitcher = Tabs;
```
Comment (bbarthec) [inline docs/components/plugins/AuthSessionElements.js:106]:
```suggestion
```

File: docs/pages/guides/authentication.md [hunk 6]
```diff
@@ -190,7 +190,7 @@ export default function App() {
 
 <AuthMethodTabSwitcher tabs={["Auth Code", "Implicit Flow"]}>
```
Comment (bbarthec) [inline docs/pages/guides/authentication.md:191]:
```suggestion
<Tabs tabs={["Auth Code", "Implicit Flow"]}>
```

File: docs/pages/guides/authentication.md [hunk 7]
```diff
@@ -190,7 +190,7 @@ export default function App() {
...
 
-<AuthCodeTab>
+<AuthMethodTab>
```
Comment (bbarthec) [inline docs/pages/guides/authentication.md:193]:
```suggestion
<Tab>
```

Comment (bbarthec) [reply docs/pages/guides/authentication.md:193]: And so on ... 😉

**Review by bbarthec — CHANGES_REQUESTED** [review]

**Review by bbarthec — APPROVED** [review]

**Pull request merged**